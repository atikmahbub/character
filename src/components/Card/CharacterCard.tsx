import { Paper, Stack, Typography, Box } from "@mui/material";
import moment from "moment";

type CardProps = {
  id: number;
  name: string;
  image: string;
  created: string;
  gender: string;
  species: string;
  handleTitleClick: (id: number) => void;
};

const CharacterCard = ({
  id,
  name,
  image,
  gender,
  species,
  created,
  handleTitleClick,
}: CardProps) => {
  return (
    <Paper elevation={1} sx={{ flex: "40%", padding: 2 }}>
      <Stack direction="row">
        <img src={image} alt="pic" width={150} />
        <Stack ml={1}>
          <Typography
            sx={{ cursor: "pointer" }}
            variant="h5"
            onClick={() => handleTitleClick(id)}
          >
            {name}
          </Typography>
          <Box display="flex" alignItems="center" gap="10px">
            <Typography variant="caption">{gender}</Typography>
            <Typography variant="caption">{species}</Typography>
          </Box>
          <Typography variant="caption">
            created: {moment(created).format("DD/MM/yyyy")}
          </Typography>
        </Stack>
      </Stack>
    </Paper>
  );
};

export default CharacterCard;
