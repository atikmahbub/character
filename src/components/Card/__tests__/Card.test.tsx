import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Card from "../CharacterCard";

const jsx = (
  <BrowserRouter>
    <Card
      id={1}
      name="Test"
      gender="Male"
      species="Human"
      handleTitleClick={() => {}}
      created="23-08-2021"
      image="Test Image"
    />
  </BrowserRouter>
);

test("Render Without Crashing", () => {
  const tree = renderer.create(jsx).toJSON();
  expect(tree).toMatchSnapshot();
});
