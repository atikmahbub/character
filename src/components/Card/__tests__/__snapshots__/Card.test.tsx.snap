// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Without Crashing 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-lclo15-MuiPaper-root"
>
  <div
    className="css-m69qwo-MuiStack-root"
  >
    <img
      alt="pic"
      src="Test Image"
      style={
        Object {
          "borderRadius": 4,
        }
      }
      width={180}
    />
    <div
      className="css-s0cucq-MuiStack-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 css-1p99d0j-MuiTypography-root"
        onClick={[Function]}
      >
        Test
      </h5>
      <div
        className="MuiBox-root css-1h2sv0j"
      >
        <span
          className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
        >
          Human
        </span>
        <span
          className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
        >
          Male
        </span>
      </div>
      <span
        className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
      >
        created: 
        Invalid date
      </span>
    </div>
  </div>
</div>
`;
